src/main/java/com/google/sps/servlets/ProjectServlet.java

```diff
@@ -0,0 +1,266 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Handles POST and GET requests for projects,
+ * allows creation and update of projects, and
+ * supports queries for projects based on various parameters.
+ */
+@WebServlet("/projects")
+public class ProjectServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Must be logged in
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/"); // placeholder: should redirect to login
```

Comment:
I forget if we are just using the gmail api for log in or something else. Basically just wondering if this should redirect to the gmail api or our 'log in' page/homepage?

Reply:
Right, since we're using the Users API, there is no login page, so I have it redirect to the homepage for now. Is there a better place to redirect to?

Reply:
The plan is to have an IntroComponent eventually, so /intro when that time comes